substrate-node/pallets/pallet-tfgrid/src/lib.rs

```diff
@@ -1081,36 +1081,37 @@ pub mod pallet {
             node_id: u32,
             node_certification: NodeCertification,
         ) -> DispatchResultWithPostInfo {
-            let account_id = ensure_signed(origin)?;
-
-            if let Some(certifiers) = AllowedNodeCertifiers::<T>::get() {
-                ensure!(
-                    certifiers.contains(&account_id),
-                    Error::<T>::NotAllowedToCertifyNode
-                );
+            // Only council/root or allowed certifiers can modify node certification
+            if !T::RestrictedOrigin::ensure_origin(origin.clone()).is_ok() {
+                let account_id = ensure_signed(origin)?;
+                if !AllowedNodeCertifiers::<T>::get()
+                    .unwrap_or(vec![])
+                    .contains(&account_id)
+                {
+                    return Err(Error::<T>::NotAllowedToCertifyNode.into());
+                }
+            }
 
-                ensure!(
-                    Nodes::<T>::contains_key(&node_id),
-                    Error::<T>::NodeNotExists
-                );
-                let mut stored_node = Nodes::<T>::get(node_id).ok_or(Error::<T>::NodeNotExists)?;
+            ensure!(
+                Nodes::<T>::contains_key(&node_id),
+                Error::<T>::NodeNotExists
+            );
+            let mut stored_node = Nodes::<T>::get(node_id).ok_or(Error::<T>::NodeNotExists)?;
```

Comment:
should we make a specific test  `fn set_certification_type_node_not_exists_fails()` to handle `NodeNotExists` error?